api/sip/__init__.py

```diff
@@ -281,20 +296,23 @@ def info_to_patrondata(cls, info, validate_password=True):
         # patron_status field will prohibit the patron from borrowing
         # books.
         status = info['patron_status_parsed']
-        block_reason = PatronData.NO_VALUE
-        for field in SIPClient.PATRON_STATUS_FIELDS_THAT_DENY_BORROWING_PRIVILEGES:
-            if status.get(field) is True:
-                block_reason = cls.SPECIFIC_BLOCK_REASONS.get(
-                    field, PatronData.UNKNOWN_BLOCK
-                )
-                if block_reason not in (PatronData.NO_VALUE,
-                                        PatronData.UNKNOWN_BLOCK):
-                    # Even if there are multiple problems with this
-                    # patron's account, we can now present a specific
-                    # error message. There's no need to look through
-                    # more fields.
-                    break
-        patrondata.block_reason = block_reason
+        if patron_blocks:
```

Comment:
This would be simpler if `patron_blocks` controlled a list of fields that was either `SIPClient.PATRON_STATUS_FIELDS_THAT_DENY_BORROWING_PRIVILEGES` or blank. You wouldn't need to indent this big section, and it would also show a path forward for a potential future expansion where the library administrator can basically customize their own list of "patron status fields that deny borrowing privileges".

Reply:
Also, at this point I think it makes sense to turn `info_to_patrondata` from a class method to an instance method, so you can access `self.patron_status_block` instead of having to pass it in.

A lot of these class methods are only that way to make testing easier, but let me know if it turns out there's a problem with this idea.

Reply:
I think these should be resolved now.